packages/material-ui/src/styles/createTypography.spec.ts

```diff
@@ -0,0 +1,15 @@
+import { createMuiTheme } from '@material-ui/core/styles';
+
+{
+  // properties of the variants can be "unset"
+  const theme = createMuiTheme({
+    typography: {
+      allVariants: {
+        fontStyle: undefined,
+      },
+    },
+  });
+
+  // $ExpectType string | undefined
```

Comment:
Was `$ExpectType string` before this change which would be to narrow given the above theme.